components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/filter/DestinationForceReloadFilter.java

```diff
@@ -0,0 +1,67 @@
+package com.sap.cloud.cmp.ord.service.filter;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientResponseException;
+
+import com.sap.cloud.cmp.ord.service.client.DestinationFetcherClient;
+import com.sap.cloud.cmp.ord.service.common.Common;
+import com.sap.cloud.cmp.ord.service.token.Token;
+import com.sap.cloud.cmp.ord.service.token.TokenParser;
+
+@Component
+public class DestinationForceReloadFilter implements Filter {
+    private static final Logger logger = LoggerFactory.getLogger(DestinationForceReloadFilter.class);
+
+    private final static String RELOAD_QUERY_PARAM = "reload";
+
+    @Value("${odata.jpa.request_mapping_path}")
+    private String odataPath;
+
+    @Autowired
+    private TokenParser tokenParser;
+
+    @Autowired
+    private DestinationFetcherClient destsFetcherClient;
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+        boolean isODataPath = Common.buildRequestPath(httpRequest).startsWith("/" + odataPath);
+        boolean reloadQueryParamIsTrue = Boolean.TRUE.toString().equals(request.getParameter(RELOAD_QUERY_PARAM));
+
+        if (isODataPath && reloadQueryParamIsTrue) {
+            Token token = tokenParser.fromRequest(httpRequest);
+            String tenant = token == null ? "" : token.extractTenant();
```

Comment:
You can use the optimization from the sensitive data filter:
```
String tenantId = (String) request.getAttribute(Common.REQUEST_ATTRIBUTE_TENANT_ID);
```
here as well.

Reply:
You are right, I think that reload will be executed relatively rarely and it seems like 3 is fine for now from my PoV.